.buildkite/commands/install-node-dependencies.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash -eu
+
+PLATFORM=$(uname -s)
+ARCHITECTURE=$(uname -m)
+NODE_VERSION=$(node --version)
+PACKAGE_HASH=$(hash_file package-lock.json)
+GUTENBERG_PACKAGE_HASH=$(hash_file gutenberg/package-lock.json)
+JETPACK_PACKAGE_HASH=$(hash_file jetpack/pnpm-lock.yaml)
+CACHEKEY="$BUILDKITE_PIPELINE_SLUG-npm-$PLATFORM-$ARCHITECTURE-node$NODE_VERSION-$PACKAGE_HASH-$GUTENBERG_PACKAGE_HASH"
+PNPM_CACHEKEY="$BUILDKITE_PIPELINE_SLUG-pnpm-$PLATFORM-$ARCHITECTURE-node$NODE_VERSION-$JETPACK_PACKAGE_HASH"
+
+echo "--- :npm: Restore cache if present"
+restore_cache "$CACHEKEY"
+restore_cache "$PNPM_CACHEKEY"
+
+echo "--- :npm: Install Node dependencies"
+npm ci --unsafe-perm --prefer-offline --no-audit --no-progress "$@"
```

Comment:
I also want to try caching the local `node_modules` etc and use `install` instead of `ci` based on rules such as:

- `trunk` builds always use `ci`, and
- deployment builds always use `ci`

But I'll leave that for another PR.